src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -105,14 +106,64 @@ public void close() {
         return request(uri, requestConfigurer.prepend(RequestSpec::post));
     }
 
-    private Action<? super RequestSpec> tracedRequestAction(final Action<? super RequestSpec> action) {
-        return action.append(requestSpec -> {
-            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(requestSpec);
+    private Action<? super RequestSpec> actionWithSpan(final Action<? super RequestSpec> action, final Span span) {
+        return action.append(request -> {
+            MethodCapturingRequestSpec captor = new MethodCapturingRequestSpec(request);
+            span.start();
             action.execute(captor);
-            HttpMethod capturedMethod = Optional.ofNullable(captor.getCapturedMethod()).orElse(HttpMethod.GET);
-            requestInterceptor
-                    .handle(requestAdapterFactory.createAdaptor(requestSpec.method(capturedMethod), capturedMethod.getName()));
+            DefaultRequestSpecNameAdapter adapter = new DefaultRequestSpecNameAdapter(captor);
+            span.name(spanNameProvider.getName(adapter));
+            span.tag(TraceKeys.HTTP_URL, adapter.getUri());
         });
     }
 
+    private void streamedResponseWithSpan(Result<StreamedResponse> response, Span span) {
+        resultWithSpan(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    private void responseWithSpan(Result<ReceivedResponse> response, Span span) {
+        resultWithSpan(response, () -> response.getValue().getStatusCode(), span);
+    }
+
+    void resultWithSpan(Result<?> result, StatusCode statusCode, Span span) {
+        if (result.isError()) {
```

Comment:
@llinder I wasn't able to write a test to exercise this code path - maybe I had the setup wrong?